### PR TITLE
restic: update to 0.8.2.

### PR DESCRIPTION
**Pull request by eichf**
updates restic to 0.8.2 and adds `version` argument functionality

### CODE REVIEW

File: srcpkgs/restic/template [hunk 1]
```diff
@@ -1,39 +1,22 @@
...
+go_import_path="github.com/${pkgname}/${pkgname}"
+go_package="${go_import_path}/cmd/${pkgname}"
+go_ldflags="-X \"main.version=${version} (Void Linux)\""
 short_desc="Backup solution written in Go"
 maintainer="Florian Eich <flrn@nrmncr.net>"
 license="BSD"
 homepage="https://restic.github.io/"
-distfiles="https://github.com/restic/restic/releases/download/v${version}/restic-${version}.tar.gz"
-checksum=1e0e0d1e7388beaa2a892c057759fdfe6a4fe915f9518c73068761f8d6d7619d
+distfiles="https://github.com/${pkgname}/${pkgname}/releases/download/v${version}/${pkgname}-${version}.tar.gz"
```
Comment (Vaelatern) [inline srcpkgs/restic/template]: Please remove your changes to this line.

File: srcpkgs/restic/template [hunk 2]
```diff
@@ -1,39 +1,22 @@
...
-version=0.8.1
-revision=2
+version=0.8.2
+revision=1
 build_style=go
-go_import_path="github.com/restic/restic"
-go_package="${go_import_path}/cmd/restic"
+go_import_path="github.com/${pkgname}/${pkgname}"
+go_package="${go_import_path}/cmd/${pkgname}"
+go_ldflags="-X \"main.version=${version} (Void Linux)\""
```
Comment (Vaelatern) [inline srcpkgs/restic/template]: Would this work with single quotes inside?

File: srcpkgs/restic/template [hunk 3]
```diff
@@ -1,39 +1,22 @@
...
 pkgname=restic
-version=0.8.1
-revision=2
+version=0.8.2
+revision=1
 build_style=go
-go_import_path="github.com/restic/restic"
-go_package="${go_import_path}/cmd/restic"
+go_import_path="github.com/${pkgname}/${pkgname}"
+go_package="${go_import_path}/cmd/${pkgname}"
```
Comment (Vaelatern) [inline srcpkgs/restic/template]: remove your changes to these lines please.

**Comment by Vaelatern** [comment]
`${pkgname}` replacements work for when the pkgname is very long. And then not in URLs unless needed.

**Comment by cr6git** [comment]
@eichf version 0.8.3 [released](https://github.com/restic/restic/releases) a couple of hours ago.

**Comment by eichf** [comment]
thanks for the hint, i'll update

**Pull request closed without merging**